File: Book Coding/Placement series AC/count_frequency.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

void countFreq(int arr[], int n){
    map <int, int> freq;

    for(int i=0; i<n; i++){
        freq[arr[i]]++;
    }
    for (auto x: freq){
        cout << x.first << " " << x.second << endl;
    }

}



int main(){
    int arr[] = { 10, 20, 20, 10, 10, 20, 5, 20 };
    int n = sizeof(arr) / sizeof(arr[0]);
    countFreq(arr, n);
    return 0;
}

File: Book Coding/Placement series AC/countfrequency.py




def countFreq(arr):
    hashi = {}
    for i in arr:
        if i in hashi:
            hashi[i] += 1
        else:
            hashi[i] = 1

    for i in hashi:
        print(i, hashi[i])

if __name__ == '__main__':
    arr = [10, 20, 20, 10, 10, 20, 5, 20]
    countFreq(arr)